src/Datadog.Trace/Tagging/TagsList.cs

```diff
@@ -13,7 +13,34 @@ internal abstract class TagsList : ITags
 
         public List<KeyValuePair<string, double>> Metrics => Volatile.Read(ref _metrics);
 
-        public List<KeyValuePair<string, string>> Tags => Volatile.Read(ref _tags);
+        public List<KeyValuePair<string, string>> GetAll()
+        {
+            var customTags = GetCustomTags();
+            var additionalTags = GetAdditionalTags();
+            var allTags = new List<KeyValuePair<string, string>>(
+                customTags?.Count ?? 0 +
+                additionalTags?.Length ?? 0);
+
+            if (customTags != null)
+            {
+                allTags.AddRange(customTags);
+            }
+
+            if (additionalTags != null)
+            {
+                foreach (var property in additionalTags)
+                {
+                    var value = property.Getter(this);
+
+                    if (value != null)
+                    {
+                        allTags.Add(new KeyValuePair<string, string>(property.Key, value));
```

Comment:
Pottential bug:
You can get duplicates if the same tag is set via `customTag` and `additionalTag`. I see in `GetTag` that `additionalTag` has priority over `customTag`. If it is an acceptable performance trade-off I would at least leave a comment that such behaviour may occur.

Reply:
AdditionalTags have priority on SetTag, so this already ensures there will be no duplicates. I didn't find any unit tests confirming it, so added one for SetTag.

Reply:
Thx